crates/hir_def/src/data.rs

```diff
@@ -123,10 +123,11 @@ impl TypeAliasData {
         let loc = typ.lookup(db);
         let item_tree = loc.id.item_tree(db);
         let typ = &item_tree[loc.id.value];
+        let type_ref = typ.type_ref.clone();
 
         Arc::new(TypeAliasData {
             name: typ.name.clone(),
-            type_ref: typ.type_ref.clone(),
+            type_ref: type_ref,
```

Comment:
```suggestion
            type_ref,
```

Reply:
I should've started from scratch XD. This is a left-over from the previous approach. Will fix shortly.